bouncer/shared/boost.ts

```diff
@@ -0,0 +1,213 @@
+  await using chainflip = await getChainflipApi();
+  const lp = keyring.createFromUri(lpUri);
+
+  assert(boostTier > 0, 'Boost fee must be greater than 0');
+
+  const observeStoppedBoosting = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:StoppedBoosting',
+    chainflip,
+    (event) =>
+      event.data.boosterId === lp.address &&
+      event.data.boostPool.asset === asset &&
+      event.data.boostPool.tier === boostTier.toString(),
+  );
+
+  const extrinsicResult = await lpSubmitExtrinsic(
+    chainflip.tx[ingressEgressPalletForChain(chainFromAsset(asset))].stopBoosting(
+      shortChainFromAsset(asset).toUpperCase(),
+      boostTier,
+    ),
+    lp,
+    errorOnFail,
+  );
+  if (!extrinsicResult.dispatchError) {
+    console.log('waiting for stop boosting event');
+    return observeStoppedBoosting;
+  }
+  console.log('Already stopped boosting');
+  return undefined;
+}
+
+/// Provides liquidity, then adds the funds to the boost pool of the given tier and returns the BoostFundsAdded event.
+export async function addBoostFunds(
+  asset: Asset,
+  boostTier: number,
+  amount: number,
+  lpUri = '//LP_BOOST',
+): Promise<Event> {
+  await using chainflip = await getChainflipApi();
+  const lp = keyring.createFromUri(lpUri);
+
+  assert(boostTier > 0, 'Boost fee must be greater than 0');
+
+  await provideLiquidity(asset, amount, true, lpUri);
+
+  const observeBoostFundsAdded = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:BoostFundsAdded',
+    chainflip,
+    (event) =>
+      event.data.boosterId === lp.address &&
+      event.data.boostPool.asset === asset &&
+      event.data.boostPool.tier === boostTier.toString(),
+  );
+
+  // Add funds to the boost pool
+  console.log(`Adding boost funds of ${amount} ${asset} at ${boostTier}bps`);
+  await lpSubmitExtrinsic(
+    chainflip.tx[ingressEgressPalletForChain(chainFromAsset(asset))].addBoostFunds(
+      shortChainFromAsset(asset).toUpperCase(),
+      amountToFineAmount(amount.toString(), assetDecimals(asset)),
+      boostTier,
+    ),
+    lp,
+  );
+
+  return observeBoostFundsAdded;
+}
+
+/// Adds boost funds to the boost pool and does a swap with boosting enabled, then stops boosting and checks the fees collected are correct.
+async function testBoostingForAsset(asset: Asset, boostFee: number, lpUri: string, amount: number) {
+  await using chainflip = await getChainflipApi();
+  console.log(`Testing boosting for ${asset} at ${boostFee}bps`);
+
+  // Start with a clean slate by stopping boosting before the test
+  const preTestStopBoostingEvent = await stopBoosting(asset, boostFee, lpUri, false);
+  assert.strictEqual(
+    preTestStopBoostingEvent?.data?.pendingBoosts?.length ?? 0,
+    0,
+    'Stopped boosting but, the test cannot start with pending boosts.',
+  );
+
+  // Add boost funds
+  console.log(`Providing liquidity for boosting`);
+  await addBoostFunds(asset, boostFee, amount, lpUri);
```

Comment:
same issue exists here as above - if a non tier fee is provided this will fail

Reply:
Also this means that we'll only be testing the boost through one pool and not through several pools? Several pools gets us more coverage with the same test duration. So, would be ideal to have one that tests that

Reply:
> Several pools gets us more coverage 

Unit tests cover this, so i think its not worth the added complexity here to test it. Maybe in the future.

Reply:
> if a non tier fee is provided this will fail

This is intended. I wanted to keep the scope of this utility function limited.